### ISSUE TITLE
Tutorial / 3D models

### ISSUE THREAD

**Issue by benoitgaudou** [gama-platform/gama#2346]
### Steps to reproduce
1. Open the Tutorials / 3D / model 02 or 03.gaml

2. Run the model

### Expected behavior
Tell us what should happen (include all relevant expectations)

### Actual behavior
A big blue cube appears and hides all the agents.
As the display seems to express that aim is to display an empty cube, there is perhaps an issue here ?
```
      	draw cube(environmentSize) color: #black empty:true;	
```

### System and version 
GAMA 1.8 preRelease, GAMA Git Mars

**Comment by agrignard** [comment]
... interesting. So indeed it seems that the empty facet is not taken in account. i am curious to see if it also happen on 1.7 (I would imagine so)

However what I don't understand is why the cube is blue (it's not even related to the fact that the agent are displayed in blue and it is also a specific blue ...

One quick fix in order at least to have a model "working" is to remove the drawing of the cube but still there is an issue with the empty facet

**Comment by benoitgaudou** [comment]
I tried on the model 3D Vizualisation / Built-in shape, to see whether the facet empty: true works on 3D geomettries. It appears that: 
it works for:
- plan ([{0,0},{size,size}],size),
- polyplan([{0,0},{size/2,size/2},{0,size}],size),
- polyhedron([{-1*size/2,0.5*size/2}, {-0.5*size/2,1*size/2}, {0.5*size/2,1*size/2}, {1*size/2,0.5*size/2},{1*size/2,-0.5*size/2},{0.5*size/2,-1*size/2},{-0.5*size/2,-1*size/2},{-1*size/2,-0.5*size/2}],size), 
- (cube(size) - cube(size / 2))

And it does not work for:
- sphere(size/2),
- line ([{0,0},{size,size}],1), 
- polyline([{0,0},{size/2,size/2},{0,size}],1), 
- cylinder(size,size),
- cube(size),
- box(size,size*1.5,size*0.5),
- pyramid(size),

Benoit

**Comment by benoitgaudou** [comment]
fixed with commit a5fe8ce2206ca5a244afc33ca3b212646e902bd4
Need some time to fix 2 more bugs:
 - when empty: true, the edges have not the same color
 - textured 3D Shape is not influenced by the wireframe preference
